src/ssl/helper.cc

```diff
@@ -43,18 +43,34 @@ class GeneratorRequest {
     CBDATA_CLASS(GeneratorRequest);
 
 public:
+    /// helper query contents plus Squid configuration ID
+    using Key = std::pair<SBuf, SquidConfig::Id::Value>;
+
+    explicit GeneratorRequest(const Key &aKey): key(aKey) {}
+
     /// adds a GeneratorRequestor
     void emplace(HLPCB *callback, void *data) { requestors.emplace_back(callback, data); }
 
-    SBuf query; ///< Ssl::Helper request message (GeneratorRequests key)
+    const Key key; ///< identifies this request in TheGeneratorRequests collection
 
     /// Ssl::Helper request initiators waiting for the same answer (FIFO).
     typedef std::vector<GeneratorRequestor> GeneratorRequestors;
     GeneratorRequestors requestors;
 };
 
-/// Ssl::Helper query:GeneratorRequest map
-typedef std::unordered_map<SBuf, GeneratorRequest*> GeneratorRequests;
+struct GeneratorRequestKeyHash
+{
+    std::size_t operator() (const GeneratorRequest::Key &pair) const {
+        return std::hash<decltype(pair.first)>()(pair.first) ^ std::hash<decltype(pair.second)>()(pair.second);
```

Comment:
The compiler fails to deduce the template type without these explicit 'decltype' specifications.